src/storage.ts

```diff
@@ -0,0 +1,94 @@
+const store = (type: 'sessionStorage' | 'localStorage') => ({
+	available: undefined as boolean | undefined,
+	storage: window[type],
+	isAvailable(): boolean | undefined {
+		const key = 'local-storage-test';
+
+		if (this.available !== undefined) {
+			return this.available;
+		}
+
+		try {
+			// to fully test, need to set item
+			// http://stackoverflow.com/questions/9077101/iphone-localstorage-quota-exceeded-err-issue#answer-12976988
+			this.storage.setItem(key, 'graun');
+			this.storage.removeItem(key);
+			this.available = true;
+		} catch (err) {
+			this.available = false;
+		}
+
+		return this.available;
+	},
+
+	get(key: string): unknown | null {
+		if (!this.available) {
+			return;
```

Comment:
?

```suggestion
			return null;
```

Reply:
The risk here is there's already code expecting `undefined` - arguably checks should be using falsey but once the api is established it's hard to move away. I might review the code in frontend to see if I can verifiy if this happens or not

Reply:
I agree it's somewhat odd but I think there's a sense to it.

```
		if (!this.available) {
			return;
		}
```
It only returns `undefined` if the storage mechanism is not available, which is different to the key not being found. I think this is a reasonable attempt to define a richer api instead of just returning falsey. `null` means there was no value for the key you requested. `undefined` means I was not able to make the request but it might still exist.